qcodes/utils/update/qcodes_updater.py

```diff
@@ -0,0 +1,127 @@
+import os
+import subprocess
+import inspect
+import shutil
+import datetime
+import warnings
+import git
+
+import qcodes
+
+def qcodes_backup(env_name: str = 'qcodes', env_backup_name: str = 'qcodes_backup'):
+    """
+    Back up QCoDeS Anaconda Python environment.
+
+    Args:
+        env_name: Name of the environment to be updated
+        env_backup_name: Name of the back up environment to be created
+
+    Raises:
+        git.exc.GitError: If working tree is dirty
+    """
+    # QCoDeS source path
+    _source = os.sep.join(qcodes.__file__.split(os.sep)[:-2])
+    # Make sure that git working tree is clean
+    repo = git.Repo(_source)
```

Comment:
We probably need code here to check if this is even a repo. E.g qcodes could be installed directly from pip